Library/Homebrew/test/utils/linkage_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "utils/linkage"
+
+RSpec.describe Utils do
+  suffix = OS.mac? ? ".dylib" : ".so"
+
+  describe "::binary_linked_to_library?", :integration_test do
```

Comment:
I can reproduce this when running the tests with `--generic`. In that case, none of the OS-specific code is loaded.

The error in CI is because `shared_library` uses the default implementation which uses the `.dylib` suffix instead of the Linux version that uses the `.so` suffix, since the Linux version is never loaded to override with `--generic`.

Also, the `dynamically_linked_libraries` implementation is OS-specific and the modules that define it aren't prepended to `Pathname` when running with `--generic`.

I think the fix is just to explicitly require macOS or Linux and not run then test for a generic OS:

```suggestion
  describe "::binary_linked_to_library?", :integration_test, :needs_linux do
```

Reply:
Thanks, I wasn't aware of the `--generic` flag 🤦.

Regarding the suggested change - this seems like it would avoid running on macOS right? Is there something available like `(:needs_linux || :needs_macos)` or `:needs_not_generic`?

Or perhaps we're comfortable with just running this on one OS, in which case I will just proceed like this.

Reply:
Unfortunately I don't think there's a simple way to specify that. It could be added easily, but it may not be worth it for just one test.

You can probably do this though:

```
[:needs_macos, :needs_linux].each do |needs_os|
  describe "...", :integration_test, needs_os do
    # ...
  end
end
```